### PR TITLE
Add tests for validators and full integration tests.

### PR DESCRIPTION
**Pull request by Mossop**

### CODE REVIEW

**Comment by Mossop** [comment]
@jamacku I want to make sure I'm testing the cases you found where Red Hat differs properly. Are there any particular public Red Hat bugs that caused you to write the fixes so I can add them to this test?

File: src/integration.test.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  await expect(
+    api.getBugs([645699, 1743832, 1749908]),
+  ).resolves.toMatchSnapshot();
+});
+
+test("Red Hat", async () => {
+  let api = new BugzillaAPI("https://bugzilla.redhat.com/");
+
+  await expect(
+    api.getBugs([1944441, 1749908, 1205830]),
```
Comment (jamacku) [inline src/integration.test.ts]: Those Bugs are great, but I would like to see also some bugs for `RHEL` (e.g. `1906064, 1730084`). Thank You!

Comment (jamacku) [reply src/integration.test.ts]: @Mossop Also I'm thinking if it would be beneficial to add some `private` bugs, for testing purposes.

Comment (Mossop) [reply src/integration.test.ts]: Private bugs would need credentials to access the bugzilla instance and then also for a proper test the private bug data would have to be in the repository. Neither of those sound great. I have a vague idea of actually starting a test bugzilla instance in CI which would solve that issue.

Comment (jamacku) [reply src/integration.test.ts]: I understand, Thanks

**Comment by jamacku** [comment]
Thank you @Mossop, great work!

**Pull request merged**